pkg/core/native/designate.go

```diff
@@ -412,10 +420,27 @@ func (s *Designate) DesignateAsRole(ic *interop.Context, r noderoles.Role, pubs
 		return fmt.Errorf("failed to update Designation role data cache: %w", err)
 	}
 
-	ic.AddNotification(s.Hash, DesignationEventName, stackitem.NewArray([]stackitem.Item{
-		stackitem.NewBigInteger(big.NewInt(int64(r))),
-		stackitem.NewBigInteger(big.NewInt(int64(ic.Block.Index))),
-	}))
+	if ic.IsHardforkEnabled(config.HFEchidna) {
+		oldNodes, _, err := s.GetDesignatedByRole(ic.DAO, r, ic.Block.Index)
+		if err != nil {
+			return fmt.Errorf("failed to get old nodes for role %d: %w", r, err)
+		}
+		newNodes, _, err := s.GetDesignatedByRole(ic.DAO, r, ic.Block.Index+1)
+		if err != nil {
+			return fmt.Errorf("failed to get new nodes for role %d: %w", r, err)
+		}
```

Comment:
It's inefficient, it will lead to one more storage seek, whereas you already have the new list of nodes available here.